core/src/main/scala/org/broadinstitute/dsp/workbench/hamm/db/WorkflowTable.scala

```diff
@@ -0,0 +1,133 @@
+    import WorkflowBinders._
+    val column = Workflow.column
+    withSQL {
+      insert.into(Workflow).namedValues(
+        column.workflowId -> workflow.workflowId,
+        column.parentWorkflowId -> workflow.parentWorkflowId,
+        column.rootWorkflowId -> workflow.rootWorkflowId,
+        column.workflowCollectionId -> workflow.workflowCollectionId,
+        column.isSubWorkflow -> workflow.isSubWorkflow,
+        column.startTime -> workflow.startTime,
+        column.endTime -> workflow.endTime,
+        column.labels -> workflow.labels,
+        column.cost -> workflow.cost)
+    }.update.apply()
+  }
+
+  def getWorkflowQuery(workflowId: WorkflowId)(implicit session: DBSession): Option[Workflow] = {
+    val e = Workflow.syntax("e")
+    withSQL {
+      select.from(Workflow as e)
+        .where.eq(e.workflowId, workflowId.id)
+    }.map(Workflow(e.resultName)).single().apply()
+  }
+
+
+  def getWorkflowCostQuery(workflowId: WorkflowId)(implicit session: DBSession): Option[Double] = {
+    val e = Workflow.syntax("e")
+    withSQL {
+      select(e.result.cost).from(Workflow as e)
+        .where.eq(e.workflowId, workflowId.id)
+    }.map(rs => rs.double(e.resultName.cost)).single().apply()
+  }
+}
+
+final case class Workflow(
+    workflowId: WorkflowId,
+    parentWorkflowId: Option[WorkflowId],
+    rootWorkflowId: Option[WorkflowId],
+    workflowCollectionId: WorkflowCollectionId,
+    isSubWorkflow: Boolean,
+    startTime: Instant,
+    endTime: Instant,
+    labels: Map[String, String],
+    cost: Double)
+
+final case class Label(key: String, value: String)
+//final case class WorkflowCost(workflowId: WorkflowId, cost: Double)
+
+object Workflow extends SQLSyntaxSupport[Workflow] {
+  override def tableName: String = "WORKFLOW"
+  import WorkflowBinders._
+  def apply(e: ResultName[Workflow])(rs: WrappedResultSet): Workflow = Workflow(
+    rs.get(e.workflowId),
+    rs.get(e.parentWorkflowId),
+    rs.get(e.rootWorkflowId),
+    rs.get(e.workflowCollectionId),
+    rs.get(e.isSubWorkflow),
+    rs.get(e.startTime),
+    rs.get(e.endTime),
+    rs.get(e.labels),
+    rs.get(e.cost)
+  )
+}
+
+object WorkflowBinders {
+  implicit val workflowIdTypeBinder: TypeBinder[WorkflowId] = new TypeBinder[WorkflowId] {
+    def apply(rs: ResultSet, label: String): WorkflowId = WorkflowId(rs.getString(label))
+    def apply(rs: ResultSet, index: Int): WorkflowId = WorkflowId(rs.getString(index))
+  }
+
+  implicit val otherValueTypeBinder: TypeBinder[Option[WorkflowId]] = {
+    TypeBinder.option[String].map(_.map(WorkflowId.apply))
+  }
+
+  implicit val WorkflowCollectionIdTypeBinder: TypeBinder[WorkflowCollectionId] = new TypeBinder[WorkflowCollectionId] {
+    def apply(rs: ResultSet, label: String): WorkflowCollectionId = WorkflowCollectionId(rs.getString(label))
+    def apply(rs: ResultSet, index: Int): WorkflowCollectionId = WorkflowCollectionId(rs.getString(index))
+  }
+
+
+  //ToDo: Make this better
```

Comment:
when are you going to make this better :)

Reply:
something tells me never